src/storage/sftp/storage.c

```diff
@@ -1098,7 +1127,15 @@ storageSftpNew(
             THROW_FMT(ServiceError, "timeout during libssh2 handshake [%d]", rc);
 
         if (rc != 0)
-            THROW_FMT(ServiceError, "libssh2 handshake failed [%d]", rc);
+        {
+            char *libSsh2ErrMsg;
+            int libSsh2ErrMsgLen;
+
+            // Get the libssh2 error message
+            rc = libssh2_session_last_error(this->session, &libSsh2ErrMsg, &libSsh2ErrMsgLen, 0);
```

Comment:
libssh2_session_last_error() is already called three times and now we are adding another call. Since it is awkward to call I think it deserves a wrapper function.

Reply:
Created storageSftpLibSsh2SessionLastError() wrapper to get libssh2 generated error message